### PR TITLE
Snapshot for standalone planner UI

### PR DESCRIPTION
**Pull request by pranavgore09**
added new stages in release.groovy for building and pushing standalone planner UI.
using deployPlannerSnapshot with a new variable set to deploy the snapshot.
Added Dockerfile.deploy.runtime that copies `./runtime/dist` to `nginx/html`

blocked by https://github.com/fabric8io/fabric8-pipeline-library/pull/378

Document is under development at  https://docs.google.com/document/d/1IuyILytkWtNRE_I7k3P8GaUylczOB0q8-aywiLgrMA8/edit#bookmark=id.d7wcetgw4wpo

### CODE REVIEW

File: Dockerfile.deploy.runtime [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM fabric8/fabric8-openshift-nginx:vd83b3a1
+MAINTAINER "Pete Muir <pmuir@bleepbleep.org.uk>"
```
Comment (debloper) [inline Dockerfile.deploy.runtime]: Nope.
He's not gonna maintain this. Either add your name or remove this line. :P

File: Jenkinsfile [hunk 2]
```diff
@@ -1,10 +1,11 @@
-@Library('github.com/fabric8io/fabric8-pipeline-library@master')
+@Library('github.com/pranavgore09/fabric8-pipeline-library@standalone-planner')
```
Comment (debloper) [inline Jenkinsfile]: If you're using some custom code, send PR to main repo and switch the DIFF back.

File: deploy/release.groovy [hunk 3]
```diff
@@ -48,6 +48,26 @@ def buildImage(imageName){
     }
 }
 
+def getStandaloneImage(imageName){
+    stage('build standalone npm') {
+        dir('runtime'){
+            container('ui'){
+               sh 'npm cache clean --force'
```
Comment (debloper) [inline deploy/release.groovy]: Not required in a fresh environment, unless you have found a reason why it's needed.

Comment (jarifibrahim) [reply deploy/release.groovy]: @debloper I don't think the environment is fresh when we rebuild the PR. At times PR build fails due to npm `integrity` errors. This command makes sure the environment is fresh before we start building the PR.

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-26 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-30 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-32 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-33 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-34 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-35 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-36 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-37 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-37 standalone planner UI is deployed and available for testing at https://fabric8-planner-fabric8-planner-standalonepr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-38 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-38 standalone planner UI is deployed and available for testing at https://fabric8-planner-fabric8-planner-standalonepr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-39 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-39 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

File: Dockerfile.deploy.runtime (hunk 1 above)
Comment (pranavgore09) [reply Dockerfile.deploy.runtime]: yes, copy-paster error!
thanks, removed in https://github.com/fabric8-ui/fabric8-planner/pull/2426/commits/e1c90ef1388ba15790c2dd04905c0d36e863e594

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-41 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-41 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-42 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-42 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-43 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-43 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-44 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-45 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:null standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:null standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:null standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:standalon-cleanup standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

File: Jenkinsfile [hunk 4]
```diff
@@ -69,10 +72,10 @@ fabric8UITemplate{
 
 // deploy a snapshot fabric8-ui pod and notify pull request of details
 if (ciDeploy){
-    def prj = 'fabric8-ui-'+ env.BRANCH_NAME
-    prj = prj.toLowerCase()
-    def route
-    timeout(time: 10, unit: 'MINUTES') {
+   timeout(time: 50, unit: 'MINUTES') {
```
Comment (kwk) [inline Jenkinsfile]: 50 minutes sounds pretty long.

Comment (pranavgore09) [reply Jenkinsfile]: yeah I understand but sometimes even 1h is less,
we face errors like below where Jenkins-slave is unavailable for the random amount of time.
```
Still waiting to schedule task
Jenkins doesn’t have label dockerfabric8_planner_27
```
In best case 10m is sufficient but current builds may take more time and timeout with a small timeout value.
`Cancelling nested steps due to timeout. Could not connect to jenkins-slave-prxp7-mp5d9 to send interrupt signal to process`

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-53 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-53 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-54 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-54 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-58 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-58 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

File: Jenkinsfile (hunk 2 above)
Comment (pranavgore09) [reply Jenkinsfile]: done  https://github.com/fabric8io/fabric8-pipeline-library/pull/378

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-60 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-60 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-62 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-64 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-68 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-68 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-69 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-69 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-70 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-70 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-76 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-76 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-77 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-77 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by pranavgore09** [comment]
Once https://github.com/fabric8io/fabric8-pipeline-library/pull/378 is merged I will replace the import line back to the original. Form `@Library('github.com/pranavgore09/fabric8-pipeline-library@standalone-planner')` to `@Library('github.com/fabric8io/fabric8-pipeline-library@master')`. Then CI should pass and we can merge it.

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-81 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-81 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-82 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2426-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@pranavgore09 fabric8/fabric8-ui:SNAPSHOT-PR-2426-82 standalone planner UI is deployed and available for testing at http://fabric8-planner-f8-plan-stdpr-2426-fabric8-planner.badger.fabric8.io

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @pranavgore09 for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin - for example exclude certain file types so if PR contains only them it won't be checked.

**Comment by pranavgore09** [comment]
closing in favor of `release from cico` already in master

**Pull request closed without merging**